src/internal-model.js

```diff
@@ -1,104 +1,121 @@
+
+      return null
     }
 
-    let relationship = this.get('relationships') && this.get('relationships')[relationName]
+    const relationship = _.result(this.get('relationships'), relationName)
+
     if (relationship == null) {
-      throw new Error('There is no relationship "' + relationName + '" in the resource.')
+      if (strict) throw new Error('There is no relationship "' + relationName + '" in the resource.')
+
+      return null
     }
 
     return relationship
   },
 
+  getRelationshipType (relationName) {
+    const relationship = _.result(this.getRelationship(relationName), 'data')
+
+    if (_.isArray(relationship)) return 'has-many'
+
+    if (relationship) return 'belongs-to'
+
+    return null
+  },
+
   hasRelated (relationName) {
-    const relationship = this.getRelationship(relationName)
-    return !!(relationship && relationship.data && relationship.data.id)
+    const relationship = this.getRelationship(relationName, false)
+
+    return !!_.result(relationship, ['data', 'id'])
   },
 
   getRelated (relationName, query) {
-    let link = this.getRelationshipLink(relationName)
-    let relType = this.getRelationshipType(relationName)
-
-    if (relType === 'unknown') {
-      return this.fetchRelated(relationName, query)
-    } else if (relType === 'has-many') {
-      let data = this.getRelationship(relationName).data
-      return this.store.getHasMany(this, link, data, query)
-    } else {
-      let {type, id} = this.getRelationship(relationName).data
-      return this.store.getBelongsTo(this, link, type, id, query)
-    }
+    const link = this.getRelationshipLink(relationName)
+
+    const related = _.result(this.getRelationship(relationName), 'data')
+
+    if (!related) return this.fetchRelated(relationName, query)
+
+    return _.isArray(related)
+      ? this.store.getHasMany(this, link, related, query)
+      : this.store.getBelongsTo(this, link, related.type, related.id, query)
   },
 
   fetchRelated (relationName, query) {
-    let link = this.getRelationshipLink(relationName)
-    let relType = this.getRelationshipType(relationName)
-
-    if (relType === 'unknown') {
-      return this.store.fetchUnknown(link, query)
-    } else if (relType === 'has-many') {
-      return this.store.fetchHasMany(this, null, link, query)
-    } else {
-      let {type, id} = this.getRelationship(relationName).data
-      return this.store.fetchBelongsTo(this, link, type, id, query)
-    }
-  },
+    const link = this.getRelationshipLink(relationName)
 
-  _getRelationForName (relationName) {
-    return this.relationships && this.relationships[relationName]
-  }
-})
+    const related = _.result(this.getRelationship(relationName), 'data')
+
+    if (!related) return this.store.fetchUnknown(link, query)
 
-export default InternalModel
+    return (_.isArray(related))
+      ? this.store.fetchHasMany(this, null, link, query)
+      : this.store.fetchBelongsTo(this, link, related.type, related.id, query)
```

Comment:
Is this breaking the "encapsulation" of store:
```
  /**
   * @private
   */
  fetchBelongsTo (owner, link, type, id, query) {
    return this.fetch(type, id, {link, query})
  }
```
Should we just mark these as public now for the `Store` class?

Reply:
I'm not sure why these would be private methods of `Store`, TBH... Only `store.getBelongsTo` calls `store.fetchBelongsTo` internally; `store.getBelongsTo` is marked `@private`; nothing calls `store.getBelongsTo` internally, only `InternalModel.getRelated`. Over-eager privacy?

Reply:
Maybe they're tagged `@private` to indicate that _outside_ callers should not use them directly? Nothing in `groveco/grove` uses them, just `getRelated`...